src/hooks/useQuery.ts

```diff
@@ -39,27 +40,34 @@ export const useQuery = <T = any, V = object>(
   const executeQuery = useCallback(
     (opts?: Partial<OperationContext>) => {
       unsubscribe();
+
       setState(s => ({ ...s, fetching: true }));
 
-      const [teardown] = pipe(
-        client.executeQuery(request, {
-          requestPolicy: args.requestPolicy,
-          ...opts,
-        }),
-        subscribe(({ data, error }) =>
-          setState({ fetching: false, data, error })
-        )
-      );
+      let teardown = noop;
+
+      if (!args.skip) {
+        [teardown] = pipe(
+          client.executeQuery(request, {
+            requestPolicy: args.requestPolicy,
+            ...opts,
+          }),
+          subscribe(({ data, error }) =>
+            setState({ fetching: false, data, error })
+          )
+        );
+      } else {
+        setState(s => ({ ...s, fetching: false }));
+      }
 
       unsubscribe = teardown;
     },
-    [request.key]
+    [request.key, args.skip, args.requestPolicy]
   );
 
   useEffect(() => {
     executeQuery();
     return unsubscribe;
-  }, [request.key]);
+  }, [request.key, args.skip]);
```

Comment:
Last question here for both of you @andyrichardson and @kitten -> should `executeQuery` be included in the dependency array here? The Hooks FAQ make a big deal about it being unsafe to exclude functions that reference `props` / `state` from the dependency array of `useEffect` calls. So basically this dep array would become just `[executeQuery]`. I'm also starting to think we might want to `useRef` `unsubscribe`. Thoughts?

Reply:
React FAQ on functions as dependencies: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
Sorry, didn't get to this :scream: Including it into the array here will probably mean that `requestPolicy` can change at any time which is generally good :+1:

Reply:
🆒 that'll come up in the next PR, gonna merge this as is 😄

Reply:
@parkerziegler - yeah I think that is necessary 👍